app/services/ping.rb

```diff
@@ -0,0 +1,67 @@
+class Ping
+  class << self
+    attr_accessor :extensions
+
+    def ping(request = nil)
+      {
+        'Foreman': {
+          database: ping_database,
+          version: SETTINGS[:version],
+          plugins: Foreman::Plugin.all,
+          smart_proxies: ping_smart_proxies(request),
+          compute_resources: ping_compute_resources(request)
+        }
+      }.merge(ping_extensions)
+    end
+
+    # Could be extended with any class which has ::ping method which returns a hash
+    def extend_with(plugin_name, extension_class)
+      self.extensions ||= {}
+      self.extensions[plugin_name] = extension_class
+    end
+
+    private
+
+    def ping_database
+      ActiveRecord::Base.connection.active? ? 'ok' : 'FAIL'
```

Comment:
Why not return a boolean value?

Reply:
Since I followed on from Katello's ping, I guess I automatically did something like https://github.com/Katello/katello/blob/39d83ea91dad4fd565330724ed6fe58bb2c9f21f/app/models/katello/ping.rb#L5. But nothing blocks me to change it in favour of a boolean value.